api/api.md

```diff
@@ -3510,7 +3510,7 @@ Authentication code for end devices.
 
 | Field | Validations |
 | ----- | ----------- |
-| `value` | <p>`string.pattern`: `^[a-zA-Z0-9]{1,32}$`</p> |
+| `value` | <p>`string.pattern`: `^[a-zA-Z0-9]{1,32}$|^$`</p> |
```

Comment:
If I'm not mistaken, we need this change since the IS store implementations do not make a difference between `null` wrapper and actual empty value, right ?

Reply:
Bun doesn't make a distinction; https://github.com/TheThingsNetwork/lorawan-stack/blob/4f42b754ca1f191a578ecd2cf0e3df82676a50ac/pkg/identityserver/bunstore/end_device_store.go#L216-L218
Gorm checks if the wrapper is null; https://github.com/TheThingsNetwork/lorawan-stack/blob/4f42b754ca1f191a578ecd2cf0e3df82676a50ac/pkg/identityserver/gormstore/end_device.go#L235-L246

We need this changes since the CAC value is not mandatory for QR codes. We pass the `EndDevice` directly and if this validation will fail if the CAC is empty.

Reply:
I'm asking this since having `claim_authentication_code.value` part of the field mask, but not having the struct itself set in the end device should have the same effect here - validation does not check unset values (that's why I mention that it should not be part of the struct).